File: RR.Frontend/src/DTOs/VendorHQ.ts
import { Address } from './Address';

export class VendorHQ {
  name: string;
  address: Address;
  taxNumber: string;
  
  constructor() {
    this.name = '';
    this.address = new Address();
    this.taxNumber = '';
  }
  
  static from(obj: any): VendorHQ {
    const v = Object.assign(new VendorHQ(), obj);
    v.address = obj.address ? Address.from(obj.address) : new Address();
    return v;
  }
  
  isEmpty(): boolean {
    return (!this.name || this.name.trim() === '') &&
      this.address.isEmpty() &&
      (!this.taxNumber || this.taxNumber.trim() === '');
  }
}

File: RR.Common/Models/VendorHQ.cs
namespace RR.Common.Models;

public class VendorHQ : IOwnable
{
    public int? Id { get; set; }
    public int? GroupId { get; set; }
    public required string Name { get; set; }
    public Address? Address { get; set; }
    public string? TaxNumber { get; set; }
}

public class VendorHQValidated : VendorHQ
{
    public Dictionary<string, string>? ValidationMessages { get; set; }
}